app/src/main/java/rick_and_morty/ui/characters/composables/CharacterList.kt

```diff
@@ -1,28 +1,54 @@
 import androidx.compose.foundation.layout.Arrangement
 import androidx.compose.foundation.lazy.LazyColumn
 import androidx.compose.foundation.lazy.items
-import androidx.compose.runtime.Composable
-import androidx.compose.runtime.LaunchedEffect
-import androidx.compose.runtime.collectAsState
+import androidx.compose.foundation.lazy.rememberLazyListState
+import androidx.compose.runtime.*
 import androidx.compose.ui.unit.dp
 import androidx.lifecycle.viewmodel.compose.viewModel
+import rick_and_morty.data.model.CharacterResultsDto
 import rick_and_morty.ui.characters.CharactersViewModel
+import rick_and_morty.ui.widgets.RickAndMortyErrorDialog
+import rick_and_morty.ui.widgets.CircularProgressBar
 
 @Composable
 fun CharacterList(charactersViewModel: CharactersViewModel = viewModel(modelClass = CharactersViewModel::class.java)) {
 
-    val characters = charactersViewModel.characters.collectAsState(emptyList())
-    LazyColumn (
-        verticalArrangement = Arrangement.spacedBy(5.dp)
-    ) {
-
-        items(characters.value) {characters ->
-            CharacterRow(characterResultsDto = characters)
-        }
-        item {
-            LaunchedEffect(true) {
+    val characters = charactersViewModel.characters.collectAsState().value
+    if (characters.isLoading)
```

Comment:
let's merge this into the `when` statement

Reply:
merging it into when statement lead to rebuild of the whole CharacterList, but if I keep it with if statement, everything works.

Reply:
Merging it into when. leads to "refresh progressbar" show separetely, but if I keep within if statement, it stays on top of the list characters.